src/petals/server/memory_cache.py

```diff
@@ -74,6 +78,8 @@ async def allocate_cache(self, *descriptors: TensorDescriptor) -> AsyncContextMa
         """
         assert os.getpid() != self.runtime_pid, "must be called by a ConnectionHandler, not runtime"
         assert all(descr.device is not None for descr in descriptors), "please specify allocated devices"
+        if self.max_alloc_timeout is not None:
+            timeout = min(timeout, self.max_alloc_timeout) if timeout is not None else self.max_alloc_timeout
```

Comment:
```suggestion
            timeout = min(timeout, self.max_alloc_timeout)
```

Let's assume timeout to be known here, otherwise we'll have defaults in two places. Notably, the default here was different from the zero default in `backend.py`, which was confusing.

Reply:
It looks like we have a misunderstanding: this is not a default. The suggested code would not support max_alloc_timeout=None

Reply:
Can we still drop `None` support here and require passing `float('inf')` for this case? The metadata's timeout is casted to float in the wrapping code anyway

Reply:
i solemnly swear to test that it works with float('inf')